### PR TITLE
feat(tutorials): adds how-to's for using hub apis in web apps

### PR DESCRIPTION
**Pull request by andrewxhill**

### CODE REVIEW

File: scripts/contributors/update.py [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+        contributors.add(dev)
+
+  # Print a copy/paste HTML block for markdown
+  print("<div class='txtl-community'>")
+  for dev in sorted(contributors, key=lambda x: x.login):
+    print(s.substitute(href=dev.html_url, src=dev.avatar_url))
+  print("</div>")
+
+if __name__ == "__main__":
+  main()
```
Comment (andrewxhill) [inline scripts/contributors/update.py:53]: just need to run this every once in a while to update our contributor grid

File: docs/tutorials/hub/web-app.md [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+
+API tokens need to be generated for every new user. Additionally, well designed tokens expire quickly, so new tokens will need to be generated for user's each time their token expires. There is two ways to handle token generate. The first is using the `getToken` method and the second is using the `getTokenChallenge`, both available in `@textile/threads-client`. Both of those methods require Hub API key and secret, so should be executed by a service only accessible to your app.
+
+**When to use the getToken method**
+
+The `getToken` method should only be used in rare cases where the application has access to the user's private key _outside of the app_. Examples might include cases where private key identities are generated server-side.
+
+[Read the tutorial on creating a token provider using getToken](gettoken-provider.md)
+
+**When to use the getTokenChallenge method**
```
Comment (carsonfarmer) [inline docs/tutorials/hub/web-app.md:47]: Niiice

File: docs/tutorials/hub/web-app.md [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+!!!info
+    In the future we will support email-based identities, allowing you to more easily use the Hub APIs with common email-based user models. [Follow progress here](https://github.com/textileio/textile/issues/216).
+
+### Creating a token endpoint
+
+API tokens need to be generated for every new user. Additionally, well designed tokens expire quickly, so new tokens will need to be generated for user's each time their token expires. There is two ways to handle token generate. The first is using the `getToken` method and the second is using the `getTokenChallenge`, both available in `@textile/threads-client`. Both of those methods require Hub API key and secret, so should be executed by a service only accessible to your app.
+
+**When to use the getToken method**
+
+The `getToken` method should only be used in rare cases where the application has access to the user's private key _outside of the app_. Examples might include cases where private key identities are generated server-side.
```
Comment (carsonfarmer) [inline docs/tutorials/hub/web-app.md:43]: In reality, this is probably overly restrictive. This method doesn't assume you have access to the key either, just that the object you are passing has a 'sign' method, which could actually abstract exactly the same way as the getTokenChallenge method. In fact, getToken simply wraps getTokenChallenge. With that in mind, I do get the need/want to have a reason for the two methods, but maybe with tempered language here?

File: docs/tutorials/static-websites/jekyll-site.md [hunk 4]
```diff
@@ -1,4 +1,4 @@
-[Jekyll](https://jekyllrb.com) is one of the most popular static website building frameworks around. Bonus, Jekyll is also open-source and free. Jekyll allows you to write your content in Markdown and then compile it to HTML with a command-line tool. When you compile your website, Jekyll will create a single folder containing your final website. This folder is what you'll want to add to a Textile Bucket.
+[Jekyll](https://jekyllrb.com) is one of the most popular static website building frameworks around. Bonus, Jekyll is also open-source and free. Jekyll allows you to write your content in Markdown and then compile it to HTML with a command-line tool. When you compile your website, Jekyll will create a single folder containing your final website. When you add that folder to a Textile Bucket, you can simultaneously publishing your Jekyll site to [IPFS](https://ipfs.io/), [IPNS](https://docs.ipfs.io/guides/concepts/ipns/), and HTTP.
```
Comment (carsonfarmer) [inline docs/tutorials/static-websites/jekyll-site.md:1]: Cool!

File: docs/tutorials/hub/web-app.md (hunk 3 above)
Comment (andrewxhill) [reply docs/tutorials/hub/web-app.md:43]: I think that's only true if you ran those methods directly from an app. Which you cannot do in this example, so here I'm referring to server-side token gen, in which case i don't think that's true. Or am I mistaken?

**Pull request merged**